### ISSUE TITLE
Serial communication

### ISSUE THREAD

**Issue by Muchwat** [SimulatedGREG/electron-vue#367]
Is there a vue-electron serialport plugin?

**Comment by SimulatedGREG** [comment]
There is not a `serialport` plugin specifically for `vue`. You can easily `import`/`require` the `serialport` library directly, and use as you normally would. There is no direct need to have a specific Vue.js plugin.

**Comment by Muchwat** [comment]
I have tried doing that but I am getting errors
Vue.use(require('serialport')) or 

import serialport from 'serialport'
Vue.use(serialport)
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7)
    at dirname (path.js:695)
    at Function.getRoot (bindings.js?59d6:151)
    at bindings (bindings.js?59d6:60)
    at eval (win32.js?e4ac:2)
    at Object.<anonymous> (renderer.js:1004)
    at __webpack_require__ (renderer.js:661)
    at fn (renderer.js:87)
    at eval (auto-detect.js?0ec6:8)
    at Object.<anonymous> 

indings/bindings.js
81:22-40 Critical dependency: the request of a dependency is an expression
./~/bindings/bindings.js
81:43-53 Critical dependency: the request of a dependency is an expression

**Comment by SimulatedGREG** [comment]
`serialport` is not a Vue plugin, therefore using with `Vue.use` will **never** work. Simply `import` the module and use it like you would any other library.

**someFile.js**
```js
import SerialPort from 'serialport'

const port = new SerialPort('/dev/tty-usbserial1', {
  baudRate: 57600
})

// now use your `port` however needed 
```

**someComponent.vue**
```html
<script>
  import SerialPort from 'serialport'

  export default {
    mounted () {
      const port = new SerialPort('/dev/tty-usbserial1', {
        baudRate: 57600
      })
      // now use your `port` however needed 
    }
  }
</script>
```

If you are looking to add `serialport` to something like `this.$serialport`, then you can do the following...

**src/renderer/main.js**
```js
import Vue from 'vue'
import SerialPort from 'serialport'

Vue.prototype.$serialPort = SerialPort

/* or even bind a port */

const port = new SerialPort('/dev/tty-usbserial1', {
  baudRate: 57600
})
Vue.prototype.$myPort = port
```
Now you can access those in any Vue component file through `this.$serialPort`/`this.$myPort`.

In conclusion, the presence of `vue` doesn't change how you handle such libraries. Yes plugins like `vue-router` and `vuex` are applied with `Vue.use`, but that doesn't mean everything else is also applied in the same manner.

**Comment by Muchwat** [comment]
Thank you SimulatedGREG, this is the best answer to my problem.

**Comment by Muchwat** [comment]
Hello SimulatedGREG, I did as you told me but the bug is still consistent. 
This is my console.

![error](https://user-images.githubusercontent.com/13133862/29743184-fa4364d2-8a94-11e7-99cc-b5ce2b0003f8.PNG)

**Comment by GoKu-gaga** [comment]
hi，when i try to require 'ffi'， i got the same error。how can i fix it？Is there any solutions？
